.github/workflows/push_branch_unix_test.yml

```diff
@@ -0,0 +1,187 @@
+name: GitHub Branch CI (unix)
+
+on:
+  push:
+    branches-ignore:
+      - master
+
+jobs:
+  pyomo-unix-tests:
+    name: ${{ matrix.TARGET }}/py${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest]
```

Comment:
Only because I was attempting to minimize the diffs between the `unix_python_matrix_test.yml` and the `push_branch_unix_test.yml`.  My rationale was that it was useful to get a spot check from all three platforms.  As this runs in a User's organization, it doesn't impact the overall throughput (and 3 builds is well under a user's concurrent quota, too).

Reply:
Agreed. I wasn't sure why we didn't do all platforms at the onset. Approved!